pkg/apiserver/domain/service/workflow.go

```diff
@@ -789,19 +789,36 @@ func (w *workflowServiceImpl) TerminateRecord(ctx context.Context, appModel *mod
 }
 
 // ResumeWorkflow resume workflow
-func ResumeWorkflow(ctx context.Context, kubecli client.Client, app *v1beta1.Application) error {
+func ResumeWorkflow(ctx context.Context, kubecli client.Client, app *v1beta1.Application, stepName string) error {
 	app.Status.Workflow.Suspend = false
 	steps := app.Status.Workflow.Steps
+	found := stepName == ""
+
 	for i, step := range steps {
 		if step.Type == wfTypes.WorkflowStepTypeSuspend && step.Phase == workflowv1alpha1.WorkflowStepPhaseRunning {
-			steps[i].Phase = workflowv1alpha1.WorkflowStepPhaseSucceeded
+			if stepName == "" {
+				steps[i].Phase = workflowv1alpha1.WorkflowStepPhaseSucceeded
+			} else if stepName == step.Name {
+				steps[i].Phase = workflowv1alpha1.WorkflowStepPhaseSucceeded
+				found = true
+				break
+			}
 		}
 		for j, sub := range step.SubStepsStatus {
 			if sub.Type == wfTypes.WorkflowStepTypeSuspend && sub.Phase == workflowv1alpha1.WorkflowStepPhaseRunning {
-				steps[i].SubStepsStatus[j].Phase = workflowv1alpha1.WorkflowStepPhaseSucceeded
+				if stepName == "" {
+					steps[i].SubStepsStatus[j].Phase = workflowv1alpha1.WorkflowStepPhaseSucceeded
+				} else if stepName == sub.Name {
+					steps[i].SubStepsStatus[j].Phase = workflowv1alpha1.WorkflowStepPhaseSucceeded
+					found = true
+					break
+				}
 			}
 		}
 	}
+	if !found {
+		return fmt.Errorf("can not find step %s", stepName)
```

Comment:
Here should return a `bcode` error, the HTTP status code should be 400.

Reply:
This function is used by api and cli, we should not wrap bcode here, maybe in the upper layer.